artificial-intelligence/ai-service/api-reference/api.yml

```
@@ -57,47 +80,1445 @@ paths:
+
+        * `ai.agent_read`
+      security:
+        - OAuth2:
+            - ai.agent_read
+      tags:
+        - Tool
+  '/ai-service/{tenant}/agentic/tools/{toolId}':
+    parameters:
+      - $ref: '#/components/parameters/path_tenant'
       - schema:
           type: string
-        name: tenant
+        name: toolId
         in: path
         required: true
+    get:
+      parameters:
+        - $ref: '#/components/parameters/fields'
+      summary: Get tool by ID
+      operationId: GET-ai-retrieve-tool
+      responses:
+        '200':
+          description: A single tool.
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/SlackNativeToolResponse'
+              examples:
+                Tool Response:
+                  $ref: '#/components/examples/SlackNativeToolResponse'
+        '401':
+          $ref: '#/components/responses/401_resp_common'
+        '403':
+          $ref: '#/components/responses/403_resp_common'
+        '404':
+          $ref: '#/components/responses/404_resp_common'
+      description: |-
+        ![Preview](https://res.cloudinary.com/saas-ag/image/upload/v1752824268/emporix/icons/preview_api1.png)
+        Retrieves the tool by given ID.
+        
+        ### Required scopes
+
+        * `ai.agent_read`
+      security:
+        - OAuth2:
+            - ai.agent_read
+      tags:
+        - Tool
+    put:
+      summary: Upsert tool
+      operationId: PUT-ai-upsert-tool
+      responses:
+        '201':
+          description: ID of created tool.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IdResponse'
+              examples:
+                Id Of Created Tool:
+                  $ref: '#/components/examples/IdResponse'
+        '204':
+          description: The resource has been successfully updated.
+        '400':
+          $ref: '#/components/responses/400_resp_common'
+        '401':
+          $ref: '#/components/responses/401_resp_common'
+        '403':
+          $ref: '#/components/responses/403_resp_common'
+        '404':
+          $ref: '#/components/responses/404_resp_common'
+        '409':
+          $ref: '#/components/responses/409_resp_common'
+      description: |-
+        ![Preview](https://res.cloudinary.com/saas-ag/image/upload/v1752824268/emporix/icons/preview_api1.png)
+        Updates or creates a specified tool by replacing all of its existing data with data from the request body.
+        If the `metadata.version` is provided then optimistic locking is enabled and version must match the version in the database.
+        
+        ### Required scopes
+
+        * `ai.agent_manage`
+      security:
+        - OAuth2:
+            - ai.agent_manage
+      requestBody:
+        $ref: '#/components/requestBodies/toolUpsertBody'
+      tags:
+        - Tool
+    patch:
+      summary: Partially update tool
```

```suggestion
      summary: Partially updating tool
```
all summaries should be in a gerund (*-ing) form